db/schema.rb

```diff
@@ -0,0 +1,63 @@
+# This file is auto-generated from the current state of the database. Instead
+# of editing this file, please use the migrations feature of Active Record to
+# incrementally modify your database, and then regenerate this schema definition.
+#
+# This file is the source Rails uses to define your schema when running `rails
+# db:schema:load`. When creating a new database, `rails db:schema:load` tends to
+# be faster and is potentially less error prone than running all of your
+# migrations from scratch. Old migrations may fail to apply correctly if those
+# migrations use external dependencies or application code.
+#
+# It's strongly recommended that you check this file into your version control system.
+
+ActiveRecord::Schema.define(version: 2020_05_01_040916) do
+
+  create_table "authentications", options: "ENGINE=InnoDB DEFAULT CHARSET=utf8mb4", force: :cascade do |t|
+    t.bigint "user_id", null: false
+    t.string "token", null: false
+    t.datetime "expires_at", null: false
+    t.datetime "created_at", precision: 6, null: false
+    t.datetime "updated_at", precision: 6, null: false
+    t.index ["token"], name: "index_authentications_on_token", unique: true
+    t.index ["user_id"], name: "index_authentications_on_user_id"
+  end
+
+  create_table "reservations", options: "ENGINE=InnoDB DEFAULT CHARSET=utf8mb4", force: :cascade do |t|
+    t.bigint "user_id", null: false
+    t.integer "event_date_id"
+    t.datetime "created_at", precision: 6, null: false
+    t.datetime "updated_at", precision: 6, null: false
+    t.index ["event_date_id"], name: "index_reservations_on_event_date_id"
+    t.index ["user_id"], name: "index_reservations_on_user_id"
+  end
+
+  create_table "users", options: "ENGINE=InnoDB DEFAULT CHARSET=utf8mb4", force: :cascade do |t|
```

Comment:
Where does the bigint user_id come from? it's referenced in reservations and authentications but I don't see it in users.

I think it would be a good idea for each row to have an bigint id field.

Reply:
Agreed. I think it's actually the default even though it's not explicitly called out in the schema file.
This is the output of the describe table
```
mysql> DESCRIBE users;
+-----------------------+--------------+------+-----+---------+----------------+
| Field                 | Type         | Null | Key | Default | Extra          |
+-----------------------+--------------+------+-----+---------+----------------+
| id                    | bigint(20)   | NO   | PRI | NULL    | auto_increment |
| user_type             | varchar(255) | NO   |     | NULL    |                |
| created_at            | datetime(6)  | NO   |     | NULL    |                |
| updated_at            | datetime(6)  | NO   |     | NULL    |                |
| first_name            | varchar(255) | YES  |     | NULL    |                |
| middle_name           | varchar(255) | YES  |     | NULL    |                |
| last_name             | varchar(255) | YES  |     | NULL    |                |
| suffix                | varchar(255) | YES  |     | NULL    |                |
| gender                | varchar(255) | YES  |     | NULL    |                |
| phone                 | varchar(255) | YES  |     | NULL    |                |
| email                 | varchar(255) | YES  |     | NULL    |                |
| address_line_1        | varchar(255) | YES  |     | NULL    |                |
| address_line_2        | varchar(255) | YES  |     | NULL    |                |
| city                  | varchar(255) | YES  |     | NULL    |                |
| state                 | varchar(255) | YES  |     | NULL    |                |
| zip_code              | varchar(255) | YES  |     | NULL    |                |
| license_plate         | varchar(255) | YES  |     | NULL    |                |
| seniors_in_household  | int(11)      | YES  |     | NULL    |                |
| adults_in_household   | int(11)      | YES  |     | NULL    |                |
| children_in_household | int(11)      | YES  |     | NULL    |                |
| permission_to_email   | tinyint(1)   | YES  |     | NULL    |                |
| permission_to_text    | tinyint(1)   | YES  |     | NULL    |                |
| date_of_birth         | date         | YES  |     | NULL    |                |
| identification_code   | varchar(255) | NO   | UNI | NULL    |                |
+-----------------------+--------------+------+-----+---------+----------------+
```